server/db/db.js

```diff
@@ -4,68 +4,114 @@
+var log = require('../logger')('server.db.db');
+var utils = require('./utils');
 
-var pool = mysql.createPool({
-  connectionLimit: config.get('db.mysql.connectionLimit'),
-  host: config.get('db.mysql.host'),
-  user: config.get('db.mysql.user'),
-  password: config.get('db.mysql.password'),
-  database: config.get('db.mysql.database')
-});
+var pool = utils.createPool();
 
-function _getConn(cb) {
-  pool.getConnection(function(err, conn) {
-    if (err) {
-      log.warn('error getting connection from pool: ' + err);
-      return cb(err);
-    }
-    cb(null, conn);
-  });
+function _trace(funcName, funcArgs) {
+  log.trace(funcName + ' called. arguments are: ' + [].join.call(funcArgs, ', '));
 }
 
 // DB API uses callbacks for the moment; TODO return promises?
+// TODO validate DB inputs from the API at some point
 module.exports = {
   createUser: function(fxaId, email, oauthToken, cb) {
-    var query = 'INSERT INTO users (fxa_id, email, oauth_token) ' +
-                'VALUES (?, ?, ?)' +
-                'ON DUPLICATE KEY UPDATE ' +
-                'fxa_id = VALUES(fxa_id), ' +
-                'email = VALUES(email), ' +
-                'oauth_token = VALUES(oauth_token)';
-    _getConn(function(err, conn) {
-      // TODO do we even want to handle pool connection errors here?
+    _trace('db.createUser', arguments);
+    var query = 'INSERT INTO users (fxaId, email, oauthToken) ' +
+                'VALUES (?, ?, ?)';
+    pool.query(query, [fxaId, email, oauthToken], function(err) {
       if (err) {
+        log.warn('error saving user: ' + err);
+        // TODO send the item to a retry queue?
         return cb(err);
       }
-      conn.query(query, [buf(fxaId), email, oauthToken], function(err) {
-        if (err) {
-          log.warn('error saving user: ' + err);
-          // TODO send the item to a retry queue?
-        }
-        conn.release();
-        cb(err);
-      });
+      log.info('created user');
+      cb(err);
     });
   },
   getUserById: function(fxaId, cb) {
-    var query = 'SELECT email, oauth_token FROM users WHERE fxa_id = ?';
-    _getConn(function(err, conn) {
-      // TODO handle conn err?
-      conn.query(query, buf(fxaId), function(err, result) {
-        if (err) {
-          log.warn('error retrieving user: ' + err);
-        }
-        conn.release();
+    _trace('db.getUserById', arguments);
+    var query = 'SELECT email, oauthToken FROM users WHERE fxaId = ?';
+    pool.query(query, fxaId, function(err, result) {
+      if (err) {
+        log.warn('error retrieving user: ' + err);
+      }
 
-        cb(err, result);
-      });
+      cb(err, result);
+    });
+  },
+  getPaginatedVisits: function(fxaId, visitId, count, cb) {
+    _trace('db.getPaginatedVisits', arguments);
+    var query = 'SELECT id, url, urlHash, title, visitedAt ' +
+                'FROM visits WHERE fxaId = ? ' +
+                'and visitedAt > (SELECT visitedAt from visits where id = ?) ' +
```

Comment:
Now I'm just being absurd, but not sure if you want to keep your SQL syntax as UPPERCASE and use `AND visitedAt > (SELECT visitedAt FROM visits WHERE id = ?)`